scene/3d/ray_cast_3d.cpp

```diff
@@ -246,10 +247,15 @@ void RayCast3D::add_exception_rid(const RID &p_rid) {
 void RayCast3D::add_exception(const Object *p_object) {
 	ERR_FAIL_NULL(p_object);
 	const CollisionObject3D *co = Object::cast_to<CollisionObject3D>(p_object);
-	if (!co) {
-		return;
+	if (co) {
+		add_exception_rid(co->get_rid());
+	} else if (Object::cast_to<GridMap>(p_object)) {
+		const GridMap *gm = Object::cast_to<GridMap>(p_object);
+		Vector<RID> rids = gm->get_static_bodies_rids();
+		for (int i = 0; i < rids.size(); i++) {
+			add_exception_rid(rids[i]);
+		}
```

Comment:
`RayCast2D` doesn't seem to do that with `TileMap`, is the situation different there?

Reply:
I haven't tested, but for what I can see exceptions won't work with TileMaps either.

Reply:
There's actually a bug report #17090

Reply:
The bug report is not for RayCast2D. But in any case, I think TileMap has the same problem.